libs/SmartSync/src/com/salesforce/androidsdk/smartsync/target/ParentChildrenSyncDownTarget.java

```diff
@@ -0,0 +1,412 @@
+        return filter;
+    }
+
+    @Override
+    protected JSONArray getRecordsFromResponseJson(JSONObject responseJson) throws JSONException {
+        JSONArray records = responseJson.getJSONArray(Constants.RECORDS);
+        for (int i=0; i<records.length(); i++) {
+            JSONObject record = records.getJSONObject(i);
+            JSONArray childrenRecords = (record.has(childrenInfo.sobjectTypePlural) && !record.isNull(childrenInfo.sobjectTypePlural)
+                    ? record.getJSONObject(childrenInfo.sobjectTypePlural).getJSONArray(Constants.RECORDS)
+                    : new JSONArray());
+            // Cleaning up record
+            record.put(childrenInfo.sobjectTypePlural, childrenRecords);
+            // XXX what if not all children were fetched
+        }
+        return records;
+    }
+
+    @Override
+    public long getLatestModificationTimeStamp(JSONArray records) throws JSONException {
+        // NB: method is called during sync down so for this target records contain parent and children
+
+        // Compute max time stamp of parents
+        long maxTimeStamp = super.getLatestModificationTimeStamp(records);
+
+        // Compute max time stamp of parents and children
+        for (int i=0; i<records.length(); i++) {
+            JSONObject record = records.getJSONObject(i);
+            JSONArray children = record.getJSONArray(childrenInfo.sobjectTypePlural);
+            maxTimeStamp = Math.max(maxTimeStamp, getLatestModificationTimeStamp(children, childrenInfo.modificationDateFieldName));
+        }
+
+        return maxTimeStamp;
+    }
+
+
+    @Override
+    protected String getDirtyRecordIdsSql(String soupName, String idField) {
+        return String.format(
+                "SELECT DISTINCT {%s:%s} FROM {%s},{%s} WHERE {%s:%s} = {%s:%s} AND ({%s:%s} = 'true' OR {%s:%s} = 'true')",
+                soupName, idField,
+                childrenInfo.soupName, soupName,
+                childrenInfo.soupName, childrenInfo.parentLocalIdFieldName,
+                soupName, SmartStore.SOUP_ENTRY_ID,
+                soupName, LOCAL,
+                childrenInfo.soupName, LOCAL);
+    }
+
+    @Override
+    protected String getNonDirtyRecordIdsSql(String soupName, String idField) {
+        return String.format(
+                "SELECT {%s:%s} FROM {%s} WHERE {%s:%s} NOT IN (%s)",
+                soupName, idField,
+                soupName,
+                soupName, SmartStore.SOUP_ENTRY_ID,
+                getDirtyRecordIdsSql(soupName, SmartStore.SOUP_ENTRY_ID)
+        );
+    }
+
+    @Override
+    public void saveRecordsToLocalStore(SyncManager syncManager, String soupName, JSONArray records) throws JSONException {
+        // NB: method is called during sync down so for this target records contain parent and children
+        SmartStore smartStore = syncManager.getSmartStore();
+        synchronized(smartStore.getDatabase()) {
```

Comment:
We were. We are doing it in other places to.
In there is that comment https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/master/libs/SmartStore/src/com/salesforce/androidsdk/smartstore/store/SmartStore.java#L259